src/app/subscriptions/data/index.ts

```diff
@@ -1,32 +1,76 @@
 import type {
   Version as PartialVersion,
   DiscoverySubscription as PartialDiscoverySubscription,
+  SubscriptionStatus as PartialSubscriptionStatus,
+  Subscription as ProtoSubscription,
 } from '@/types/index.d'
 
+const acknowledgements = [
+  'responsesSent',
+  'responsesAcknowledged',
+  'responsesRejected',
+] as const
+type Acknowledgements = {
+  [key in (typeof acknowledgements)[number]]: number
+}
+
 export type Version = PartialVersion
-export type Subscription = {
-  disconnectTime?: string
-  status: {
-    lastUpdateTime: string
-  }
+
+export type PartialSubscription = {
+  status: PartialSubscriptionStatus
   version?: Version
-}
+} & ProtoSubscription
+
 export type DiscoverySubscription = PartialDiscoverySubscription
-export type SubscriptionCollection<T extends { version?: any }> = {
-  subscriptions: T[]
-  connectedSubscription?: T
-  version?: T['version']
-}
-export type DiscoverySubscriptionCollection = SubscriptionCollection<DiscoverySubscription>
 
+export const Subscription = {
+  fromObject<T extends PartialSubscription>(item: T) {
+    return {
+      $raw: item,
```

Comment:
We usually name this key `config`, correct?

Reply:
Yeah I bumped into this during the work. A `subscription` already has a `config` property, so we can't use that name 😬 .

We wrote about this eventuality somewhere and said we could use `$raw` if we ever bumped into this (and in other places we already use `$raw`). We could change the places where we use `config` for this so that they all use `$raw` really so its all consistent.

Lemme see if I can find where we wrote about this

Reply:
I see 🙂 I like consistency, so I'd vote for making use of `$raw` as this key is more likely not going to appear in the api payloads 🙂